app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -295,7 +292,9 @@ public CommCareSession getCurrentSession() {
 
     public AndroidSessionWrapper getCurrentSessionWrapper() {
         if (sessionWrapper == null) {
-            throw new SessionUnavailableException();
+            // TODO PLM: should be able to init this so it is never null.
+            // Need to find the correct place after the currentApp is set.
+            throw new SessionStateUninitException("CommCare user session isn't available");

```

Comment:
Making a distinction between login sessions and commcare state sessions, and the exceptions that are thrown when they aren't available.